pallets/staking/src/pallet/mod.rs

```diff
@@ -0,0 +1,1034 @@
+		/// `ErasTotalStake`, `ErasStartSessionIndex`,
+		/// `StakingLedger.claimed_rewards`.
+		///
+		/// Must be more than the number of eras delayed by session.
+		/// I.e. active era must always be in history. I.e. `active_era >
+		/// current_era - history_depth` must be guaranteed.
+		///
+		/// If migrating an existing pallet from storage value to config value,
+		/// this should be set to same value or greater as in storage.
+		///
+		/// Note: `HistoryDepth` is used as the upper bound for the `BoundedVec`
+		/// item `StakingLedger.claimed_rewards`. Setting this value lower than
+		/// the existing value can lead to inconsistencies in the
+		/// `StakingLedger` and will need to be handled properly in a migration.
+		/// The test `reducing_history_depth_abrupt` shows this effect.
+		#[pallet::constant]
+		type HistoryDepth: Get<u32>;
+
+		/// The overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Handler for the unbalanced reduction when slashing a staker.
+		type Slash: OnUnbalanced<NegativeImbalanceOf<Self>>;
+
+		/// Handler for the unbalanced increment when rewarding a staker.
+		/// NOTE: in most cases, the implementation of `OnUnbalanced` should modify the total
+		/// issuance.
+		type Reward: OnUnbalanced<PositiveImbalanceOf<Self>>;
+
+		/// Number of eras that staked funds must remain bonded for.
+		#[pallet::constant]
+		type BondingDuration: Get<EraIndex>;
+
+		/// Number of eras that slashes are deferred by, after computation.
+		///
+		/// This should be less than the bonding duration. Set to 0 if slashes
+		/// should be applied immediately, without opportunity for intervention.
+		#[pallet::constant]
+		type SlashDeferDuration: Get<EraIndex>;
+
+		/// The origin which can cancel a deferred slash. Root can always do this.
+		type SlashCancelOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// The maximum number of `unlocking` chunks a [`StakingLedger`] can
+		/// have. Effectively determines how many unique eras a staker may be
+		/// unbonding in.
+		///
+		/// Note: `MaxUnlockingChunks` is used as the upper bound for the
+		/// `BoundedVec` item `StakingLedger.unlocking`. Setting this value
+		/// lower than the existing value can lead to inconsistencies in the
+		/// `StakingLedger` and will need to be handled properly in a runtime
+		/// migration. The test `reducing_max_unlocking_chunks_abrupt` shows
+		/// this effect.
+		#[pallet::constant]
+		type MaxUnlockingChunks: Get<u32>;
+
+		/// A hook called when any staker is slashed. Mostly likely this can be a no-op unless
+		/// other pallets exist that are affected by slashing per-staker.
+		type OnStakerSlash: sp_staking::OnStakerSlash<Self::AccountId, Self::CurrencyBalance>;
+
+		// Some parameters of the benchmarking.
+		//type BenchmarkingConfig: BenchmarkingConfig;
+
+		// Weight information for extrinsics in this pallet.
+		//TODO finish benchmarks.
+		type WeightInfo: WeightInfo;
```

Comment:
Can you put these in `pallets/staking/src/weights.rs` and `pallets/staking/src/benchmarking.rs` when you come to it? 

I've updated the parity checking script in #883 to be able to discover the staking pallet and extract extrinsics out of it but it will still look for the benchmarks and weights in fixed locations. See https://github.com/gluwa/creditcoin/actions/runs/3911712880/jobs/6685414925

FTR the script will need to be adjusted a bit more I think.

Reply:
I can copy the weights for substrate's staking. What exactly do you need from benchmarking.rs? So far, I can't compile benchmarks because the pallet needs to be 'in use' by the runtime to benchmark it.